src/java.base/unix/native/libnio/ch/Net.c

```diff
@@ -671,6 +671,16 @@ Java_sun_nio_ch_Net_joinOrDrop4(JNIEnv *env, jobject this, jboolean join, jobjec
         n = setsockopt(fdval(env,fdo), IPPROTO_IP, opt, optval, optlen);
     }
 #endif
+#ifdef _AIX
+    // workaround AIX bug where IP_ADD_MEMBERSHIP fails intermittently
+    if (n < 0 && errno == EAGAIN) {
+        int countdown = 3;
+        while (n < 0 && errno == EAGAIN && countdown > 0) {
+            n = setsockopt(fdval(env,fdo), IPPROTO_IP, opt, optval, optlen);
+            countdown--;
+        }
+    }
+#endif
```

Comment:
drive-by comment, would it not make sense to then wait some milliseconds before the next try?

Reply:
I am unsure if this would be better - I have seen test failing about 2-3 times a day in our nightly tests. With a direct retry I have seen tests failing only every 3-5 days - with a second retry it failed only one time in 3 weeks. Therefore I decided to go for 3 retries without waiting. 
Another possibility would be to wait for an increasing number of milliseconds - like 0, 10, 20ms in each try. But I am not sure if this is worth it.
What do you think?

Reply:
Do we know what the underlying problem is? We are waiting on the kernel? Or something outside?

Reply:
(If you want to push this, go ahead; I don't want to hold up this patch)

Reply:
> Do we know what the underlying problem is? We are waiting on the kernel? Or something outside?
I have no further information why this errno is returned, only that EAGAIN ist not specified by AIX as an errno for  setsockopt.

Reply:
> (If you want to push this, go ahead; I don't want to hold up this patch)
Thanks - I will integrate it now.